operator/scheduler/control_plane_test.go

```diff
@@ -0,0 +1,288 @@
+/*
+Copyright (c) 2024 Seldon Technologies Ltd.
+
+Use of this software is governed by
+(1) the license included in the LICENSE file or
+(2) if the license included in the LICENSE file is the Business Source License 1.1,
+the Change License after the Change Date as each is defined in accordance with the LICENSE file.
+*/
+
+package scheduler
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/seldonio/seldon-core/apis/go/v2/mlops/scheduler"
+
+	mlopsv1alpha1 "github.com/seldonio/seldon-core/operator/v2/apis/mlops/v1alpha1"
+	"github.com/seldonio/seldon-core/operator/v2/pkg/constants"
+)
+
+func TestSendWithTimeout(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	type test struct {
+		name      string
+		sleepTime time.Duration
+		err       error
+		isErr     bool
+		isExpired bool
+	}
+
+	fn := func(err error) error {
+		time.Sleep(5 * time.Millisecond)
+		return err
+	}
+
+	tests := []test{
+		{
+			name:      "simple",
+			sleepTime: 10 * time.Millisecond,
+			err:       nil,
+			isErr:     false,
+			isExpired: false,
+		},
+		{
+			name:      "timeout",
+			sleepTime: 1 * time.Millisecond,
+			err:       nil,
+			isErr:     true,
+			isExpired: true,
+		},
+		{
+			name:      "error",
+			sleepTime: 10 * time.Millisecond,
+			err:       fmt.Errorf("error"),
+			isErr:     true,
+			isExpired: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			hasExpired, err := execWithTimeout(func() error {
+				return fn(test.err)
+			}, test.sleepTime)
+			g.Expect(hasExpired).To(Equal(test.isExpired))
+			if test.isErr {
+				g.Expect(err).ToNot(BeNil())
+			} else {
+				g.Expect(err).To(BeNil())
+			}
+		})
+	}
+}
+
+func TestControlPlaneEvents(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	type test struct {
+		name                            string
+		existing_resources              []client.Object
+		expected_requests_pipelines     []*scheduler.LoadPipelineRequest
+		expected_requests_models        []*scheduler.LoadModelRequest
+		expected_requests_models_unload []*scheduler.UnloadModelRequest
+		expected_requests_servers       []*scheduler.ServerNotify
+		expected_requests_experiments   []*scheduler.StartExperimentRequest
+	}
+	now := metav1.Now()
+
+	// note expected state is derived in the test, maybe we should be explictl about it in the future
```

Comment:
```suggestion
	// note expected state is derived in the test, maybe we should be explicit about it in the future
```

Reply:
Most of the expected state is still declared, right? I assume this is about the deleted pipeline and experiment?

Reply:
thats a stale comment, removed now.